src/mtcp/mtcp_restart.c

```diff
@@ -668,6 +668,37 @@ static void unmap_memory_areas_and_restore_vdso(RestoreInfo *rinfo)
     }
     MTCP_ASSERT(vdso == rinfo->vdsoStart);
   }
+#if defined(__i386__)
```

Comment:
The new code needs to go inside the previous `if  (vdsoStart != NULL) { ... }` block.

You also need to do the same i386 specific trick to the `if  (vvarStart != NULL) { ... }` block.